planner/core/testdata/enforce_mpp_suite_out.json

```diff
@@ -974,5 +974,85 @@
         "Warn": null
       }
     ]
+  },
+  {
+    "Name": "TestMPPSingleDistinct3Stage",
+    "Cases": [
+      {
+        "SQL": "set @@tidb_allow_mpp=1;set @@tidb_enforce_mpp=1;",
+        "Plan": null,
+        "Warn": null
+      },
+      {
+        "SQL": "EXPLAIN select count(distinct c) from t;",
+        "Plan": [
+          "TableReader_34 1.00 root  data:ExchangeSender_33",
+          "└─ExchangeSender_33 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection_27 1.00 mpp[tiflash]  Column#7",
+          "    └─HashAgg_28 1.00 mpp[tiflash]  funcs:sum(Column#9)->Column#7",
+          "      └─ExchangeReceiver_32 1.00 mpp[tiflash]  ",
+          "        └─ExchangeSender_31 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "          └─HashAgg_28 1.00 mpp[tiflash]  funcs:count(distinct test.t.c)->Column#9",
+          "            └─ExchangeReceiver_30 1.00 mpp[tiflash]  ",
+          "              └─ExchangeSender_29 1.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.t.c, collate: binary]",
```

Comment:
how about adding a test about distinct col with a case_insensitive collation?

Reply:
only for this case, that the ExchangeSender here only needs to send every hash key once may be a small optimal point (SQL itself don't care about the other duplicated rows, but the following case is not)

Reply:
I think HashAgg_26 already removed the duplicates, right?

Reply:
not sure